zerver/tests/test_upload.py

```diff
@@ -37,6 +39,34 @@ def destroy_uploads():
         shutil.rmtree(settings.LOCAL_UPLOADS_DIR)
 
 class FileUploadTest(AuthedTestCase):
+
+    def test_rest_endpoint(self):
+        # type: () -> None
+        """
+        Tests the /api/v1/user_uploads api endpoint. Here a single file is uploaded
+        and downloaded using a username and api_key
+        """
+        user_profile = get_user_profile_by_email('hamlet@zulip.com')
+        username = 'hamlet@zulip.com'
+        api_key = user_profile.api_key
+        fp = StringIO("zulip!")
+        fp.name = "zulip.txt"
+
+        # Upload file via API
+        auth_headers = {'HTTP_AUTHORIZATION': 'Basic ' + base64.b64encode(username + ':' + api_key)}
```

Comment:
You can use `**self.api_auth(email)` for this rather than constructing it by hand.